apps/src/code-studio/assets/SpriteUpload.js

```diff
@@ -1,41 +1,107 @@
 import React from 'react';
 
-export default class SpriteUpload extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      file: null
-    };
+const styles = {
+  uploadStatusMessage: {
+    fontSize: 20
+  },
+  uploadFailure: {
+    color: 'red'
   }
+};
+
+export default class SpriteUpload extends React.Component {
+  state = {
+    fileData: null,
+    filePreviewURL: '',
+    filename: '',
+    category: '',
+    uploadStatus: {
+      success: null,
+      message: ''
+    }
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
 
-  handleSubmit(event) {}
+    return fetch(
+      `/api/v1/animation-library/level_animations/${this.state.filename}`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'image/png'
+        },
+        body: this.state.fileData
+      }
+    )
+      .then(response => {
+        let responseMessage = response.ok
+          ? 'Image Successfully Uploaded'
+          : `Error(${response.status}: ${response.statusText})`;
+        this.setState({
+          uploadStatus: {success: response.ok, message: responseMessage}
+        });
+      })
+      .catch(err => {
+        this.setState({uploadStatus: {success: false, message: err}});
```

Comment:
i think `err` here will be an `Error`, not a string -- we could do the same to transform the error into a string as on line 40 (or, ideally, create a helper method that transforms an Error into our string message!)